stac_api/runtime/src/app.py

```diff
@@ -82,10 +83,26 @@ async def lifespan(app: FastAPI):
     collections_get_request_model=collections_get_request_model,
     items_get_request_model=items_get_request_model,
     response_class=ORJSONResponse,
-    middlewares=[Middleware(CompressionMiddleware), Middleware(ValidationMiddleware)],
+    middlewares=[Middleware(ValidationMiddleware), Middleware(TenantFilterMiddleware)],
     router=APIRouter(route_class=LoggerRouteHandler),
 )
-app = api.app
+
+if (
+    auth_settings.openid_configuration_url
+    and auth_settings.openid_configuration_internal_url
+):
+    # Use stac-auth-proxy when authentication is enabled, which it will be for production envs
+    app = configure_app(
+        api.app,
+        upstream_url=(api_settings.custom_host + (api_settings.root_path or "")),
+        oidc_discovery_url=str(auth_settings.openid_configuration_url),
+        oidc_discovery_internal_url=str(auth_settings.openid_configuration_url),
+        default_public=True,
+        root_path=api_settings.root_path,
```

Comment:
There are a few more settings that we can make use of:
```suggestion
        root_path=api_settings.root_path,
        openapi_spec_endpoint=api_settings.openapi_spec_endpoint,
```